toolchain/check/call.cpp

```diff
@@ -169,62 +212,15 @@ auto PerformCall(Context& context, SemIR::LocId loc_id, SemIR::InstId callee_id,
   }
 
   if (callee_specific_id->has_value()) {
-    auto generic_callee_id = callee_id;
-
-    // Strip off a bound_method so that we can form a constant specific callee.
-    auto bound_method = context.insts().TryGetAs<SemIR::BoundMethod>(callee_id);
-    if (bound_method) {
-      generic_callee_id = bound_method->function_decl_id;
-    }
-
-    // Form a specific callee.
-    if (callee_function.self_type_id.has_value()) {
-      // This is an associated function in an interface; the callee is the
-      // specific function in the impl that corresponds to the specific function
-      // we deduced.
-      callee_id = GetOrAddInst(
-          context, context.insts().GetLocId(generic_callee_id),
-          SemIR::SpecificImplFunction{
-              .type_id = GetSingletonType(
-                  context, SemIR::SpecificFunctionType::SingletonInstId),
-              .callee_id = generic_callee_id,
-              .specific_id = *callee_specific_id});
-    } else {
-      // This is a regular generic function. The callee is the specific function
-      // we deduced.
-      callee_id = GetOrAddInst(
-          context, context.insts().GetLocId(generic_callee_id),
-          SemIR::SpecificFunction{
-              .type_id = GetSingletonType(
-                  context, SemIR::SpecificFunctionType::SingletonInstId),
-              .callee_id = generic_callee_id,
-              .specific_id = *callee_specific_id});
-    }
-
-    // Add the `self` argument back if there was one.
-    if (bound_method) {
-      callee_id = GetOrAddInst<SemIR::BoundMethod>(
-          context, loc_id,
-          {.type_id = bound_method->type_id,
-           .object_id = bound_method->object_id,
-           .function_decl_id = callee_id});
-    }
+    callee_id = ResolveCalleeSpecific(context, loc_id, callee_id,
```

Comment:
Maybe `BuildCalleeSpecificFunction`, given that this is only for the function case. `Resolve` vs `Build` I could go either way, but the resolve step here is a bit indirect these days (it happens as a consequence of the constant evaluation of the `SpecificFunction`) so `Build` seems maybe a bit clearer.

Reply:
Renamed, also changed the comment a little to mirror.